tests/units/units.cpp

```diff
@@ -2760,3 +2760,32 @@ TEST(Units, validateConnectionsWithBaseUnits)
 
     EXPECT_EQ(size_t(0), validator->issueCount());
 }
+
+TEST(Units, equivalentUnitsMatchingBuiltinUnits)
+{
+    libcellml::UnitsPtr units = libcellml::Units::create("kelvin");
+    units->addUnit(libcellml::Units::StandardUnit::KELVIN);
+
+    libcellml::UnitsPtr unitsClone = units->clone();
+
+    EXPECT_TRUE(libcellml::Units::compatible(units, unitsClone));
+}
+
+TEST(Units, equivalentUnitsMatchingBuiltinUnitsFromVariable)
+{
+    libcellml::ModelPtr model = libcellml::Model::create("unit_compatible_test");
+
+    libcellml::VariablePtr variable = libcellml::Variable::create("T");
+    libcellml::UnitsPtr units = libcellml::Units::create("kelvin");
+    units->addUnit(libcellml::Units::StandardUnit::KELVIN);
+
+    libcellml::UnitsPtr unitsClone = units->clone();
+
+    model->addUnits(units);
+    variable->setUnits(units);
+
+    libcellml::VariablePtr variableParam = variable->clone();
+    variableParam->setInitialValue(293);
```

Comment:
Not needed for this test.

Reply:
[2b8c06d](https://github.com/cellml/libcellml/pull/1145/commits/2b8c06d41a5400303f6171ed91d5d69ff035bcba)